node-registrar/pkg/db/db.go

```diff
@@ -134,3 +142,29 @@ func (db Database) Close() error {
 	}
 	return nil
 }
+
+// Transaction executes operations within a database transaction
+func (db *Database) Transaction(fn func(tx *gorm.DB) error) error {
+	return db.gormDB.Transaction(fn)
+}
+
+// MigrateNodeLastSeen updates the LastSeen field for existing nodes that don't have it set
+func (db Database) MigrateNodeLastSeen() (int64, error) {
```

Comment:
we have a file migrate.go we can move this to

Reply:
Sure